quickwit-metastore/migrations/2021-08-29-122131_create_splits/up.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TABLE splits (
+    split_id VARCHAR(50) PRIMARY KEY, 
+    split_state INTEGER NOT NULL, 
+    num_records BIGINT NOT NULL, 
+    size_in_bytes BIGINT NOT NULL, 
+    start_time_range BIGINT,
+    end_time_range BIGINT,
+    generation BIGINT NOT NULL,
+    update_timestamp BIGINT NOT NULL, 
```

Comment:
Having a strict schema is can be great, but we will have more than one implementation, but as I commented on notion, we should rely on JSON here for most of this data.

We want to have as a field:

split_id,
index_id,
split_state,
start_time_range
end_time_range

Possibly tags as well, (which would then be an array).
The rest should be in a JSON field.

Reply:
I left the following columns in the table and added a new `json` column.
The `json` column now contains a JSON string of `SplitMetadataAndFooterOffsets`.
This is so that the string in the `json` column can be easily deserialized to `SplitMetadataAndFooterOffsets`.

```
CREATE TABLE splits (
    split_id VARCHAR(50) PRIMARY KEY,
    split_state VARCHAR(30) NOT NULL,
    start_time_range BIGINT,
    end_time_range BIGINT,
    tags TEXT NOT NULL,
    json TEXT NOT NULL,
    index_id VARCHAR(50) NOT NULL,

    FOREIGN KEY(index_id) REFERENCES indexes(index_id)
);
```